src/Features/LanguageServer/Protocol/Features/TodoComments/TodoCommentOptionsStorage.cs

```diff
@@ -2,15 +2,24 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
+using System.Collections.Generic;
+using System.Collections.Immutable;
 using Microsoft.CodeAnalysis.Options;
 
 namespace Microsoft.CodeAnalysis.TodoComments
 {
     internal static class TodoCommentOptionsStorage
     {
-        public static readonly Option2<string> TokenList = new("TodoCommentOptions", "TokenList", TodoCommentOptions.Default.TokenList);
+        public static readonly Option2<ImmutableArray<string>> TokenList = new(
```

Comment:
We don't have any way to say the underlying storage is an ImmutableArray<> but expose it as the nicer types? That way we could avoid the caching in the AbstractTodoCommentsIncrementalAnalyzer?

Reply:
Not without introducing a custom option storage. We can do that later.